### PR TITLE
Correct to_python for phonenumbers.phonenumber.PhoneNumber type values

### PR DESCRIPTION
**Pull request by tdobrovolny**
Correct issue #77 

Function to_python for values with type of
phonenumbers.phonenumber.PhoneNumber must use merge_from method of this
class. It is not possible simply cast value because constructur does not
support it (its first argument is country_code).

Add tests for it.

### CODE REVIEW

File: phonenumber_field/phonenumber.py [hunk 1]
```diff
@@ -87,7 +87,8 @@ def to_python(value):
             phone_number = PhoneNumber(raw_input=value)
     elif (isinstance(value, phonenumbers.phonenumber.PhoneNumber) and
           not isinstance(value, PhoneNumber)):
-        phone_number = PhoneNumber(value)
```
Comment (tdobrovolny) [inline phonenumber_field/phonenumber.py]: Cast that does not work and cause Error

File: phonenumber_field/phonenumber.py [hunk 2]
```diff
@@ -87,7 +87,8 @@ def to_python(value):
...
+        phone_number = PhoneNumber()
```
Comment (tdobrovolny) [inline phonenumber_field/phonenumber.py]: Create empty field

File: phonenumber_field/phonenumber.py [hunk 3]
```diff
@@ -87,7 +87,8 @@ def to_python(value):
...
+        phone_number.merge_from(value)
```
Comment (tdobrovolny) [inline phonenumber_field/phonenumber.py]: Merge required value to empty field

File: phonenumber_field/tests.py [hunk 4]
```diff
@@ -1,14 +1,13 @@
 # -*- coding: utf-8 -*-
 
-from django.test.testcases import TestCase
```
Comment (tdobrovolny) [inline phonenumber_field/tests.py]: Reorder imports, hope it never mind :-)

File: phonenumber_field/tests.py [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
-from phonenumber_field.modelfields import PhoneNumberField
-from phonenumber_field.phonenumber import PhoneNumber
-from phonenumber_field.validators import to_python
 from django.conf import settings
+from django.db import models
+from django.test.testcases import TestCase
+from phonenumber_field.modelfields import PhoneNumberField
+from phonenumber_field.phonenumber import PhoneNumber, to_python
+from phonenumbers import phonenumberutil
+import phonenumbers
```
Comment (tdobrovolny) [inline phonenumber_field/tests.py]: import whole phonenumbers

File: phonenumber_field/tests.py [hunk 6]
```diff
@@ -1,14 +1,13 @@
...
 
-from phonenumber_field.modelfields import PhoneNumberField
-from phonenumber_field.phonenumber import PhoneNumber
-from phonenumber_field.validators import to_python
 from django.conf import settings
+from django.db import models
+from django.test.testcases import TestCase
+from phonenumber_field.modelfields import PhoneNumberField
+from phonenumber_field.phonenumber import PhoneNumber, to_python
+from phonenumbers import phonenumberutil
```
Comment (tdobrovolny) [inline phonenumber_field/tests.py]: shortcut for phonenumberutil

File: phonenumber_field/tests.py [hunk 7]
```diff
@@ -60,14 +59,18 @@ def test_objects_with_same_number_are_equal(self):
             MandatoryPhoneNumber.objects.create(
                 phone_number=number_string).phone_number
             for number_string in self.equal_number_strings]
-        self.assertTrue(all(is_number_match(n, numbers[0]) == MatchType.EXACT_MATCH
```
Comment (tdobrovolny) [inline phonenumber_field/tests.py]: change for import whole phonenumbers

File: phonenumber_field/tests.py [hunk 8]
```diff
@@ -60,14 +59,18 @@ def test_objects_with_same_number_are_equal(self):
...
+        self.assertTrue(
+            all(phonenumbers.is_number_match(n, numbers[0]) ==
+                phonenumbers.MatchType.EXACT_MATCH for n in numbers))
 
     def test_field_returns_correct_type(self):
         model = OptionalPhoneNumber()
         self.assertEqual(model.phone_number, None)
         model.phone_number = '+49 176 96842671'
         self.assertEqual(type(model.phone_number), PhoneNumber)
+        model.phone_number = phonenumberutil.parse(
```
Comment (tdobrovolny) [inline phonenumber_field/tests.py]: Test that phonenumbers.phonenumber.PhoneNumber type can be assignet

File: phonenumber_field/tests.py [hunk 9]
```diff
@@ -60,14 +59,18 @@ def test_objects_with_same_number_are_equal(self):
...
+            self.test_number_1, keep_raw_input=True)
+        self.assertEqual(type(model.phone_number), PhoneNumber)
```
Comment (tdobrovolny) [inline phonenumber_field/tests.py]: And return correct type

File: phonenumber_field/phonenumber.py (hunk 2 above)
Comment (thedrow) [inline phonenumber_field/phonenumber.py]: Why did you change this?

Comment (tdobrovolny) [inline phonenumber_field/phonenumber.py]: Because constructor of ```PhoneNumber``` class (which you call by ```PhoneNumber(value)``` on line 90), has as the first positional argument ```country_code```. It is the reason why is not possible simply cast ```value```. So I create "empty" ```PhoneNumber``` instance instead and then, on next line (91), set it to the same ```value``` by ```merge_from```, which correctly handle any type of PhoneNumber.

**Comment by thedrow** [comment]
Thanks for your contribution.
